src/react/components/buttons/IconButton.jsx

```diff
@@ -87,31 +103,45 @@ const getConfirmProps = (type) => {
     };
 };
 
-const ThemeButton = ThemeConsumer(({ theme, bordered, iconSize, ...rest }) => {
-    let color = theme?.color?.accent;
-    if (color && Oskari.util.isDarkColor(color)) {
-        color = theme?.color?.primary;
+const ThemeButton = ThemeConsumer(({
+    theme,
+    type,
+    bordered = false,
+    disabled = false,
+    icon = type ? getPredefinedIcon(type) : null,
+    iconSize = 16,
+    color = TYPE_COLORS[type],
+    className = '',
+    active = false,
+    ...rest
+}) => {
+    if (type && !className.includes(`t_${type}`)) {
+        className = `${className} t_${type}`;
     }
-    if (!color) {
-        color = '#ffd400';
-    } else if (Oskari.util.isDarkColor(color)) {
-        color = getColorEffect(color, EFFECT.LIGHTEN);
+    let hover = theme?.color?.accent;
+    if (hover && Oskari.util.isDarkColor(hover)) {
+        hover = theme?.color?.primary;
     }
```

Comment:
I don't quite understand this logic. If hover (accent color) is dark -> hover is changed to primary color? Is this correct? Then it's checked again for being dark in the next else if and a lighten effect is applied?

Reply:
Changed for this pr only variable name (color -> hover). I think that the goal is to get light hover color based on theme. Not sure if hover should be lighten accent if both colors are dark (now it's lighten primary).
`
let hover = theme?.color?.accent || theme?.color?.primary || '#ffd400';
if (Oskari.util.isDarkColor(color)) {
    color = getColorEffect(color, EFFECT.LIGHTEN);
}
`
Not sure if only accent color is used in theme. If it's both or none, could be:
`
let hover = theme?.color?.accent;
if (hover && Oskari.util.isDarkColor(hover)) {
    hover = getColorEffect(hover, EFFECT.LIGHTEN);
} else if (!hover ) {
    hover = '#ffd400'
}
`
First contains useless isDark check for #ffd400 if theme isn't defined

Reply:
Checked how MapButton uses hover and found this:
https://github.com/oskariorg/oskari-frontend/blob/master/src/react/theme/ThemeHelper.js#L36